#795 增加kafka sasl权限认证；SASLMechanism 目前仅支持PLAIN、SCRAM-SHA-256、SCRAM-SHA-512

renheqiang opened the pull request:
增加kafka sasl权限认证；SASLMechanism 目前仅支持PLAIN、SCRAM-SHA-256、SCRAM-SHA-512

renheqiang committed:
增加kafka sasl权限认证；SASLMechanism 目前仅支持PLAIN、SCRAM-SHA-256、SCRAM-SHA-512

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/alibaba/MongoShake?pullRequest=820) <br/>All committers have signed the CLA.

renheqiang commented:
**Description**
增加kafka开启了sasl权限认证时的连接，在配置文件可配置如下参数
当tunnel为kafka，增加了

```
# 是否开启sasl，默认为false
tunnel.kafka.sasl.enable = true 
# 配置用户名密码 
tunnel.kafka.sasl.auth = {username}@{password} 
# 配置sasl.mechanism的方式，目前支持PLAIN、SCRAM-SHA-256、SCRAM-SHA-512。
tunnel.sasl.mechanism = SCRAM-SHA-256 
```

zhangst commented:
感谢贡献代码，会在下一个版本中合并

Neustradamus commented:
Linked to:
- https://github.com/scram-sasl/info/issues/1

renheqiang committed:
增加kafka压缩参数；
1、tunnel.kafka.compression，默认none，支持gzip,snappy,lz4,zstd
2、tunnel.kafka.producer.maxmessagebytes，默认18M,由于kafka传输不是单挑写入，同一批次可能会超过18M，故将此参数开放出来

renheqiang commented:
增加kafka参数；
1、tunnel.kafka.compression，默认none，支持压缩gzip,snappy,lz4,zstd
2、tunnel.kafka.producer.maxmessagebytes，默认18M,由于kafka传输不是单条写入，同一批次可能会超过18M，故将此参数开放出来

zhongli-james commented:
由于是历史PR，考虑到冲突，我先手动合并了，会在v2.8.7中发布。再次感谢贡献！~

Neustradamus commented:
@zhongli-james: Why have you closed this @renheqiang PR?

zhongli-james commented:
> @zhongli-james: Why have you closed this @renheqiang PR?

Feature will be released in v2.8.7. (Sorry that my previous reply was only in Chinese)

Neustradamus commented:
@zhongli-james: The PR has been manually merged?

zhongli-james commented:
> @zhongli-james: The PR has been manually merged?

Yes.
